lib/api/api_csrf.go

```diff
@@ -178,3 +178,14 @@ func (m *csrfManager) load() {
 		m.tokens = append(m.tokens, s.Text())
 	}
 }
+
+func hasValidAPIKeyHeader(r *http.Request, validator apiKeyValidator) bool {
+	if key := r.Header.Get("X-API-Key"); key != "" {
+		return validator.IsValidAPIKey(key)
+	}
+	if auth := r.Header.Get("Authorization"); strings.HasPrefix(strings.ToLower(auth), "bearer ") {
+		bearerToken := auth[len("bearer "):]
+		return validator.IsValidAPIKey(bearerToken)
+	}
+	return false
```

Comment:
Perhaps allow either to be valid?

I.e., first being wrong but second one correct still suggests success?

Reply:
Sure, there. Keeping the X-API-Key check first ensures some sort of compatibility if someone, today, makes requests with X-API-Key set and some random bearer token from somewhere else.